base_decorator.rb

```diff
@@ -0,0 +1,12 @@
+class BaseDecorator < Nameable
+  attr_accessor :nameable
```

Comment:
- [ ] Kindly import the `Nameable` class as requested by the previous reviewer.

        require_relative './nameable'